subsys/bluetooth/common/log.h

```diff
@@ -42,45 +41,6 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL);
 #define BT_WARN(fmt, ...) LOG_WRN(fmt, ##__VA_ARGS__)
 #define BT_INFO(fmt, ...) LOG_INF(fmt, ##__VA_ARGS__)
 
-#if defined(CONFIG_BT_ASSERT_VERBOSE)
-#define BT_ASSERT_PRINT(test) __ASSERT_LOC(test)
-#define BT_ASSERT_PRINT_MSG(fmt, ...) __ASSERT_MSG_INFO(fmt, ##__VA_ARGS__)
-#else
-#define BT_ASSERT_PRINT(test)
-#define BT_ASSERT_PRINT_MSG(fmt, ...)
-#endif /* CONFIG_BT_ASSERT_VERBOSE */
-
-#if defined(CONFIG_BT_ASSERT_PANIC)
-#define BT_ASSERT_DIE k_panic
-#else
-#define BT_ASSERT_DIE k_oops
-#endif /* CONFIG_BT_ASSERT_PANIC */
-
-#if defined(CONFIG_BT_ASSERT)
-#define BT_ASSERT(cond)                          \
-	do {                                     \
-		if (!(cond)) {                   \
-			BT_ASSERT_PRINT(cond);   \
-			BT_ASSERT_DIE();         \
-		}                                \
-	} while (0)
-
-#define BT_ASSERT_MSG(cond, fmt, ...)                              \
-	do {                                                       \
-		if (!(cond)) {                                     \
-			BT_ASSERT_PRINT(cond);                     \
-			BT_ASSERT_PRINT_MSG(fmt, ##__VA_ARGS__);   \
-			BT_ASSERT_DIE();                           \
-		}                                                  \
-	} while (0)
-#else
-#define BT_ASSERT(cond) __ASSERT_NO_MSG(cond)
-#define BT_ASSERT_MSG(cond, msg, ...) __ASSERT(cond, msg, ##__VA_ARGS__)
-#endif/* CONFIG_BT_ASSERT*/
-
-#define BT_HEXDUMP_DBG(_data, _length, _str) \
-		LOG_HEXDUMP_DBG((const uint8_t *)_data, _length, _str)
```

Comment:
You are removing this with the argument that it just casts the `_data` to `uint8_t *`, and that it isn't needed. 

Why are we keeping the `BT_<LOG_LEVEL>` macros and removing this? 

If you are unhappy about the `(const uint8_t *)`, can't we just remove that from the macro? Seems it is unneeded since you just replaced it with `LOG_HEXDUMP_DBG`. 

I don't see the point of removing just this macro, and your argument is very thin IMO

Reply:
check the PR message, we want to remove the BT_<LOG_LEVEL>, this is just the first PR

Reply:
Ah, sorry, missed that we wanted to remove the entire `log.h` file :) So we are removing all the `BT_<LOG_LEVEL>` 

Are we keeping the `BT_ASSERT` macros? If so, why? If not, then it doesn't make much sense to move them in this PR IMO.

Reply:
we'd like to remove the `BT_ASSERT` macros too in a future PR yeah. But it's unplanned right now, we're just focusing on log because that's what causing the most pain.

We were discussing that having a nicer way of checking params would be great. The `CHECKIF` macro seems a good fit for that since it can easily be promoted to an assert during development/testing. When we do this, we'll most likely remove `BT_ASSERT` in the same move.

Reply:
> we'd like to remove the BT_ASSERT macros too in a future PR yeah

That should be fine, but careful with the Controller's `LL_ASSERT`, which doesn't play by `__ASSERT`'s rules.

Reply:
We will remove the `BT_<LOG_LEVEL>` macros in the next PR in favor of standard Zephyr logging functions.

The BT logging mechanism is getting in the way of using the 'info' log level. We are basically forbidden from using `BT_INFO` because that level is forced on for all modules when `CONFIG_BT_DEBUG_LOG=y`. Using standard logging will give the developer control over the log level on a per-module basis. BT logging does not have any advantage over standard logging, so we are just removing the whole shebang.

The removal of `BT_HEXDUMP_DBG` in particular is motivated in the commit message (copied below).

> Replace unnecessary `BT_HEXDUMP_DBG` with `LOG_HEXDUMP_DBG`.
>
> `BT_HEXDUMP_DBG` did an extra cast to `const uint8_t *` on its first
> argument. This was probably the reason it was introduced. But the
> current definition of `LOG_HEXDUMP_DBG` does this cast for us.

I.e. `BT_HEXDUMP_DBG` is exactly the same as the Zephyr standard `LOG_HEXDUMP_DBG`, so we see no need to keep it.